pkg/reconciler/serverlessservice/serverlessservice.go

```diff
@@ -206,36 +206,49 @@ func (r *reconciler) updateStatus(sks *netv1alpha1.ServerlessService) (*netv1alp
 
 func (r *reconciler) reconcilePublicService(ctx context.Context, sks *netv1alpha1.ServerlessService) error {
 	logger := logging.FromContext(ctx)
+
+	// We need to program the public service to direct traffic to the Activator
+	// until we have serving backends. Thus, regardless of the SKS serving mode,
+	// we should configure service to be backed by Activator, until there are ready pods.
```

Comment:
I'd like to better understand why this is necessary.  Let's chat tomorrow.

Reply:
Sure. I'd be happy to simplify this, but did not find a way . how :)

Reply:
As discussed in person. #3873.